### ISSUE TITLE
(>1) bundle and "composite products": issue in total amount importing.

### ISSUE THREAD

**Issue by alexletizia** [WPCV/wpcv-woo-civi-integration#23]
To offer "real products" through Woocommerce, on CiviCRM we create a price set for both simple products and products with variants (by setting the relative price field for each variant). The price fields are set as "text / numeric quantity" so to receive any amount from Woocommerce. It works.

However: 

- if, for example, I buy two shirts of the same size (whether they are simple or variable products) even at two different times but in the same order, Woocomerce correctly returns the value of the single shirt x2 and the value is correctly imported on CiviCRM;

- if I create "**composite products**" or "**bundles**" (that show the CiviCRM Settings panel on) even if on CiviCRM I have created a specific field for each of them, **if I buy only one** it works;
but **if I buy more than one** (whether they are identical or slightly different but pointing to the same price field; even at two different times but in the same order,) the total amount on CiviCRM is non imported.

![Immagine 2022-02-23 153738](https://user-images.githubusercontent.com/40597363/155340912-6bee4573-fddb-4fb8-9d4b-02113332502e.jpg)

Do you think it depends on the composites? https://it.wordpress.org/plugins/wpc-composite-products/

or maybe we miss some house?

or does it depend on how "Integrate CiviCRM with WooCommerce" works and you think it can be solved?

**Comment by christianwach** [comment]
> if I create "composite products" or "bundles"

@alexletizia Are you are using a plugin to enable these "bundles"? There is currently only specific support for the Product Types that ship with WooCommerce out-of-the-box. 

FYI the CiviCRM Settings for Woo tab allows you to restrict the Product Types that the CiviCRM Options tab appears on so that your Users don't get confused.

**Comment by alexletizia** [comment]
> > if I create "composite products" or "bundles"
> 
> @alexletizia Are you are using a plugin to enable these "bundles"? There is currently only specific support for the Product Types that ship with WooCommerce out-of-the-box.
> 
> FYI the CiviCRM Settings for Woo tab allows you to restrict the Product Types that the CiviCRM Options tab appears on so that your Users don't get confused.

Yes, we use the plugin https://it.wordpress.org/plugins/wpc-composite-products/

So a bundle (or composite product) can just be a "grouped product" (unfortunately with limited functions) in Woocommerce out of the box, right?

**Comment by christianwach** [comment]
> If I buy more than one (whether they are identical or slightly different but pointing to the same price field; even at two different times but in the same order,) the total amount on CiviCRM is non imported.

It sounds like your Product Bundle plugin is building the Order in a way that isn't compatible with this plugin when this is the case. I'd have to look into what's involved in providing support for it, but you could start by debugging what's going on [in this method](https://github.com/WPCV/wpcv-woo-civi-integration/blob/7239d485c3c03323cbe86c10fd95287185ca077a/includes/classes/class-woo-civi-products.php#L232) to see if your Item(s) and Product(s) contains the correct data.

**Comment by alexletizia** [comment]
We do not have such high PHP skills to be able to do debugging. :(

However, we can tell you that we have partially solved the problem by using "dummy bundles" through this plugin https://wordpress.org/plugins/wpc-grouped-product/ (smart grouped products). 

It is not the same thing but, at least, it generates "single transactions" (simple products or products with varietions) and the full amount is imported.
